From e8bce0c294ce3decc36c56b9d1d4dc336538e9ab Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 11 Jul 2016 11:54:31 +0100
Subject: [PATCH] thermald: thd_cdev_modem: ensure private variables in class
 are initialized

throttling and online are not initialized, so set the appropriately in the
constructor.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_cdev_modem.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/thd_cdev_modem.cpp b/src/thd_cdev_modem.cpp
index a079a1b..77138a7 100644
--- a/src/thd_cdev_modem.cpp
+++ b/src/thd_cdev_modem.cpp
@@ -128,6 +128,9 @@ cthd_cdev_modem::ofono_signal_handler (DBusConnection *conn,
 
 cthd_cdev_modem::cthd_cdev_modem(unsigned int _index, std::string control_path)
 		: cthd_cdev(_index, "") {
+	throttling = false;
+	online = false;
+
 	if (control_path.length() > 0)
 		modem_path.assign(control_path);
 	else
-- 
2.8.1

